### PR TITLE
feat: Add Spinner component (#341)

### PR DESCRIPTION
**Pull request by MarinaRomanova**
## 📋 Changes description
- Add `Spinner` component
- Depricate `CircularProgressIndicatorIndeterminate`

## 📸 Screenshots
Generated by paparazzi in this PR files

[SpinnerDemo.webm](https://github.com/adevinta/spark-android/assets/36896406/2ebb861b-17ab-475c-b667-da02341ab931)

## 🗒️ Other info
It is possible that `CircularProgressIndicator` with a fixed progress will be updated in the future, but for now Material Design wrapper can still be used.

### CODE REVIEW

**Review by shkschneider — APPROVED** [review]

**Review by shkschneider — APPROVED** [review]

**Review by shkschneider — APPROVED** [review]

**Review by shkschneider — APPROVED** [review]

**Review by shkschneider — APPROVED** [review]

**Review by shkschneider — APPROVED** [review]

**Review by shkschneider — APPROVED** [review]

**Review by shkschneider — DISMISSED** [review]

**Review by turolopezsanabria — APPROVED** [review]
✅ ok design review

File: spark/src/main/kotlin/com/adevinta/spark/components/progress/SpinnerDefaults.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ * SOFTWARE.
+ */
+
+package com.adevinta.spark.components.progress
+
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+
+public object SpinnerDefaults {
+    public val CircularSize: SpinnerSize = SpinnerSize.Medium
```
Comment (ThibaultChariere) [inline spark/src/main/kotlin/com/adevinta/spark/components/progress/SpinnerDefaults.kt:29]: Maybe I'm missing something but I don't feel this name is very explicit. I don't understand how Circular can be a size given they are alway circular 🤔

File: spark/src/main/kotlin/com/adevinta/spark/components/progress/SpinnerDefaults.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    internal val IndicatorStrokeWidth = 2.dp
+
+    // Each rotation is 1 second
```
Comment (vyguera) [inline spark/src/main/kotlin/com/adevinta/spark/components/progress/SpinnerDefaults.kt:32]: You could simply use SECOND_IN_MILLIS constant and get rid of the comment.

File: spark/src/main/kotlin/com/adevinta/spark/components/progress/Spinner.kt [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    val rotation = transition.animateFloat(
+        0f,
+        SpinnerDefaults.BaseRotationAngle,
+        infiniteRepeatable(
+            animation = tween(
+                durationMillis = SpinnerDefaults.RotationDuration,
+                easing = LinearEasing,
+            ),
+        ),
+        label = "Spinner BAse Rotation",
```
[the remark this answers is not in the document]
Comment (MarinaRomanova) [reply spark/src/main/kotlin/com/adevinta/spark/components/progress/Spinner.kt:83]: Indeed! Thanks! I correct it with a next commit

File: spark/src/main/kotlin/com/adevinta/spark/components/progress/Spinner.kt [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+        isBackgroundVisible = isBackgroundVisible,
+    )
+}
+
+@Preview(
+    group = "Spinner",
+    name = "Spinner Medium",
+)
+@Composable
+internal fun PreviewSpinnerMedium(
```
[the remark this answers is not in the document]
Comment (MarinaRomanova) [reply spark/src/main/kotlin/com/adevinta/spark/components/progress/Spinner.kt:154]: Yes, it will be applied in the next commit

**Pull request merged**